service/src/main/java/com/codeforcommunity/processor/MapProcessorImpl.java

```diff
@@ -122,4 +159,124 @@ public NeighborhoodGeoResponse getNeighborhoodGeoJson() {
             .collect(Collectors.toList());
     return new NeighborhoodGeoResponse(features);
   }
+
+  private <T> List<T> mergeSorted(List<T> l1, List<T> l2, Comparator<T> comparator) {
+    if (l1.isEmpty()) {
+      return l2;
+    }
+    if (l2.isEmpty()) {
+      return l1;
+    }
+    T element1 = l1.get(0);
+    T element2 = l2.get(0);
+    List<T> newList = new ArrayList<>();
+    if (comparator.compare(element1, element2) < 0) {
+      newList.add(element1);
+      newList.addAll(mergeSorted(l1.subList(1, l1.size()), l2, comparator));
+      return newList;
+    }
+    newList.add(element2);
+    newList.addAll(mergeSorted(l1, l2.subList(1, l2.size()), comparator));
+    return newList;
+  }
+
+  @Override
+  public SiteGeoResponse getSiteGeoJson() {
+    List<
+            Record9<
+                Integer,
+                Boolean,
+                Double,
+                String,
+                Timestamp,
+                String,
+                String,
+                BigDecimal,
+                BigDecimal>>
+        nonNullUserRecords =
+            this.db
+                .select(
+                    SITES.ID,
+                    SITE_ENTRIES.TREE_PRESENT,
+                    SITE_ENTRIES.DIAMETER,
+                    SITE_ENTRIES.SPECIES,
+                    SITE_ENTRIES.UPDATED_AT,
+                    USERS.USERNAME,
+                    SITES.ADDRESS,
+                    SITES.LAT,
+                    SITES.LNG)
+                .from(SITES)
+                .innerJoin(SITE_ENTRIES)
+                .on(SITES.ID.eq(SITE_ENTRIES.SITE_ID))
+                .innerJoin(USERS)
+                .on(SITE_ENTRIES.USER_ID.eq(USERS.ID))
+                .where(
```

Comment:
So this checks that you're returning only the latest site entries? Does this work?

It might be easier to do a sub query that retrieves a table of the latest site entry per site and the join on that, but if this works I'm okay with it.

Reply:
👍 on this, might be an easier approach but if this works I'm okay with it

Reply:
Yep I tested it and it works